src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -241,4 +230,110 @@ private void emitGlobalSetter(Constant segmentConstant, Constant vhConstant, Str
             """);
         decrAlign();
     }
+
+    public String emitGlobalSegment(MemoryLayout layout, String javaName, String nativeName, Declaration declaration) {
+        String mangledName = mangleName(javaName, MemorySegment.class);
+        appendIndentedLines(STR."""
+            private static final MemorySegment \{mangledName} = RuntimeHelper.lookupGlobalVariable(\"\{nativeName}\", \{layoutString(0, layout)});
+
+            """);
+        if (declaration != null) {
+            emitDocComment(declaration);
+        }
+        appendIndentedLines(STR."""
+            \{MEMBER_MODS} MemorySegment \{mangledName}() {
+                return RuntimeHelper.requireNonNull(\{mangledName}, \"\{javaName}\");
+            }
+            """);
+        return mangledName;
+    }
+
+    private String emitVarLayout(MemoryLayout layout, String javaName) {
+        String mangledName = mangleName(javaName, MemoryLayout.class);
+        appendIndentedLines(STR."""
+            private static final MemoryLayout \{mangledName} = \{layoutString(0, layout)};
+
+            \{MEMBER_MODS} MemoryLayout \{mangledName}() {
+                return \{mangledName};
+            }
```

Comment:
Now that we no longer have the constants classes, I wonder if we should just do away with the getter here and just expose the layout field.

I noticed that we are also already exposing the FunctionDescriptor in a functional interface class as a field, as part of this patch,

Reply:
I wondered that also. I think that, for the context of this PR, I'd like to avoid changes to the shape of the generated code. Should be easy to fix as a followup change. As observed offline, some getters perform extra validation (e.g. null checks).